### PR TITLE
Added MultiLineString support for @turf/point-on-line

### LINKED ISSUE #835: MultiLineString support for point-on-line
It would be nice if the point-on-line module could accept either a LineString or a MultiLineString parameter.

### PR DESCRIPTION
**Pull request by stebogit**
Fixes #835 

Still the module returns a single `point`, the closest of all the `lines` of the `MultiLineString`.
Is it correct?

CC: @jbccollins @DenisCarriere

### CODE REVIEW

File: packages/turf-point-on-line/index.js [hunk 1]
```diff
@@ -44,24 +47,22 @@ var lineString = helpers.lineString;
...
-    var coords;
-    if (line.type === 'Feature') {
-        coords = line.geometry.coordinates;
-    } else if (line.type === 'LineString') {
-        coords = line.coordinates;
-    } else {
-        throw new Error('input must be a LineString Feature or Geometry');
+module.exports = function (lines, pt, units) {
+    // validation
+    if (lines.type !== 'Feature' && lines.type !== 'LineString' && lines.type !== 'MultiLineString' && lines.geometry.type !== 'LineString' && lines.geometry.type !== 'MultiLineString') {
```
Comment (DenisCarriere) [inline packages/turf-point-on-line/index.js]: Condensed this validation step to this:

```js
var type = (lines.geometry) ? lines.geometry.type : lines.type;
if (type !== 'LineString' && type !== 'MultiLineString') {
    throw new Error('lines must be LineString or MultiLineString');
}
```

File: packages/turf-point-on-line/index.js [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+var coordAll = meta.coordAll;
 var lineString = helpers.lineString;
+
 /**
- * Takes a {@link Point} and a {@link LineString} and calculates the closest Point on the LineString.
+ * Takes a {@link Point} and a {@link LineString} and calculates the closest Point on the (Multi)LineString.
  *
  * @name pointOnLine
- * @param {Feature<LineString>} line line to snap to
+ * @param {Feature<LineString>|Feature<MultiLineString>} lines lines to snap to
```
Comment (DenisCarriere) [inline packages/turf-point-on-line/index.js]: To support both Feature & Geometry, so far the JSDocs have been declared like this:

```js
/**
 * @param {Geometry|Feature<LineString|MultiLineString>} lines lines to snap to
 * @param {Geometry|Feature<Point>|number[]} pt point to snap from
```

File: packages/turf-point-on-line/index.js (hunk 1 above)
Comment (stebogit) [inline packages/turf-point-on-line/index.js]: 🤦‍♂️ of course! Dumb me 😅

**Comment by DenisCarriere** [comment]
Using happening when the point is between the MultiLineString.
image

File: packages/turf-point-on-line/index.js (hunk 1 above)
Comment (DenisCarriere) [inline packages/turf-point-on-line/index.js]: No no... it's just simpler to read this way.

Comment (stebogit) [inline packages/turf-point-on-line/index.js]: exactly!
simpler => better!!  😃

Comment (DenisCarriere) [inline packages/turf-point-on-line/index.js]: Agreed! 😄

**Comment by stebogit** [comment]
Fixed with `flattenEach`.

screen shot 2017-07-12 at 11 52 17 pm

Added also a more complex test:
screen shot 2017-07-12 at 11 58 07 pm

**Comment by DenisCarriere** [comment]
@stebogit Awesome! Looks good to me

**Pull request merged**